### PR TITLE
Static build and use explicit bounds

### PR DESCRIPTION
**Pull request by amarrella**
Breaking change.

This adds a `static` flag to the executables that can be used for static builds.
Dependencies should do the same.

This will primarily reduce the maintenance burden of having to maintain a dedicated docker image and improve build times by cache-sharing with other builds that use the same `lts` docker image.

Also starting to set explicit bounds to prepare publication to hackage (we still need to find a name that doesn't clash with the other runtime currently in hackage)

### CODE REVIEW

**Comment by amarrella** [comment]
@louoso fyi :)

File: README.md [hunk 1]
```diff
@@ -6,15 +6,43 @@ A custom runtime client for AWS Lamda written in Haskell.
 
 Examples of how to use this client can be found in the `examples` directory.
 
-## Prerequisites ##
+## Creating a lambda based on this library
+To create a lambda that is based on this library, using `stack`:
+
+1. Add it to your extra-deps this repository
```
Comment (barambani) [inline README.md]:
```suggestion
1. Add this repository to your extra-deps
```

**Comment by amarrella** [comment]
I'll be waiting before merging this so that more eyes looked at it.
You can experiment using it by pointing at the commit hash in this PR.

**Comment by louoso** [comment]
I can definitely see the appeal, but I don't know enough to say whether or not you'll run into any weird issues. Good luck!

**Pull request merged**